mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/world/stats/VersionedWorldStatsProvider.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * FlintMC
+ * Copyright (C) 2020-2021 LabyMedia GmbH and contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package net.flintmc.mcapi.v1_15_2.world.stats;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import net.flintmc.framework.eventbus.EventBus;
+import net.flintmc.framework.eventbus.event.subscribe.PostSubscribe;
+import net.flintmc.framework.eventbus.event.subscribe.Subscribe;
+import net.flintmc.framework.eventbus.event.subscribe.Subscribe.Phase;
+import net.flintmc.framework.inject.implement.Implement;
+import net.flintmc.mcapi.event.DirectionalEvent.Direction;
+import net.flintmc.mcapi.server.event.PacketEvent;
+import net.flintmc.mcapi.server.event.ServerConnectEvent;
+import net.flintmc.mcapi.server.event.ServerDisconnectEvent;
+import net.flintmc.mcapi.world.stats.StatsCategory;
+import net.flintmc.mcapi.world.stats.WorldStatType;
+import net.flintmc.mcapi.world.stats.WorldStats;
+import net.flintmc.mcapi.world.stats.WorldStatsProvider;
+import net.flintmc.mcapi.world.stats.event.PlayerStatsUpdateEvent;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.StatsScreen;
+import net.minecraft.network.play.client.CClientStatusPacket;
+import net.minecraft.network.play.client.CClientStatusPacket.State;
+import net.minecraft.network.play.server.SStatisticsPacket;
+import net.minecraft.stats.StatisticsManager;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+@Singleton
+@Implement(value = WorldStatsProvider.class, version = "1.15.2")
+public class VersionedWorldStatsProvider implements WorldStatsProvider {
+
+  private final VersionedWorldStatsMapper mapper;
+  private final StatsScreenShadow shadow;
+
+  private final EventBus eventBus;
+  private final PlayerStatsUpdateEvent event;
+
+  private CompletableFuture<WorldStats> pendingRequest;
+  private WorldStats lastStats;
+
+  @Inject
+  private VersionedWorldStatsProvider(VersionedWorldStatsMapper mapper, EventBus eventBus) {
+    this.mapper = mapper;
+
+    this.eventBus = eventBus;
+    this.event = new PlayerStatsUpdateEvent() {
```

Comment:
Please create a concrete implementation in the internal source set if this takes no parameters instead of using anonymous class here.